app/services/claims/fee_calculator/calculate.rb

```diff
@@ -0,0 +1,161 @@
+require 'ostruct'
+
+# Service to calculate the total "price/bill" for a given fee.
+# Note that this price will require input from different attributes
+# on the claim and may require input from different CCCD fees
+# to be consolidated/munged.
+#
+module Claims
+  module FeeCalculator
+    Response = Struct.new(:success?, :data, :errors, :message, keyword_init: true)
+    Data = Struct.new(:amount, keyword_init: true)
+
+    class Calculate
+      delegate  :earliest_representation_order_date,
+                :agfs?,
+                :case_type,
+                :offence,
+                to: :claim
+
+      # TODO: check if these can all be made private
+      attr_reader :claim, :options, :fee_type, :advocate_category, :quantity, :current_page_fees
+
+      def initialize(claim, options)
+        @claim = claim
+        @options = options
+      end
+
+      def call
+        setup(options)
+        response(true, amount)
+      rescue StandardError => err
+        Rails.logger.error("error: #{err.message}")
+        response(false, err, I18n.t('fee_calculator.calculate.amount_unavailable'))
+      end
+
+      private
+
+      def setup(options)
+        @fee_type = Fee::BaseFeeType.find(options[:fee_type_id])
+        @advocate_category = options[:advocate_category] || claim.advocate_category
+        @quantity = options[:quantity] || 1
+        @current_page_fees = options[:fees].values
+      rescue StandardError
```

Comment:
❓ What's the intent of this? We're catching all exceptions that happen here and raising a very generic error message, loosing the original exception details in the process.

More so, the `call` method is also catching all `StandardError` exceptions (again), which means, we're just rescue/catch exceptions multiple times.

Looking at the code it seems there's no obvious reason to have the rescue over here (but I lack context, so, not sure).

Reply:
This was to specifically handle the situation where insufficient details have been passed to this class to calculate a fee. In particular if, for example, you do not select an advocate category before selecting a fee type you would get an error from API through no fault of the user. I had intended to refactor this to some degree.